diff --git "a/versions\\1.20.1\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\BasaltColumnsFeature.java" "b/versions\\1.21.6\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\BasaltColumnsFeature.java"
index cd4346b..812297e 100644
--- "a/versions\\1.20.1\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\BasaltColumnsFeature.java"
+++ "b/versions\\1.21.6\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\BasaltColumnsFeature.java"
@@ -120,7 +120,7 @@ public class BasaltColumnsFeature extends Feature<BasaltColumnsFeatureConfig> {
 
 	@Nullable
 	private static BlockPos moveUpToAir(WorldAccess world, BlockPos.Mutable mutablePos, int distance) {
-		while (mutablePos.getY() < world.getTopY() && distance > 0) {
+		while (mutablePos.getY() <= world.getTopYInclusive() && distance > 0) {
 			distance--;
 			BlockState blockState = world.getBlockState(mutablePos);
 			if (CANNOT_REPLACE_BLOCKS.contains(blockState.getBlock())) {
