diff --git "a/versions\\1.20.1\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\ConfiguredFeatures.java" "b/versions\\1.21.6\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\ConfiguredFeatures.java"
index 7c60b97..5ae4da8 100644
--- "a/versions\\1.20.1\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\ConfiguredFeatures.java"
+++ "b/versions\\1.21.6\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\ConfiguredFeatures.java"
@@ -55,7 +55,7 @@ public class ConfiguredFeatures {
 	}
 
 	public static RegistryKey<ConfiguredFeature<?, ?>> of(String id) {
-		return RegistryKey.of(RegistryKeys.CONFIGURED_FEATURE, new Identifier(id));
+		return RegistryKey.of(RegistryKeys.CONFIGURED_FEATURE, Identifier.ofVanilla(id));
 	}
 
 	public static void register(
