diff --git "a/versions\\1.20.1\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\DiskFeature.java" "b/versions\\1.21.6\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\DiskFeature.java"
index 11eb40e..1a1c219 100644
--- "a/versions\\1.20.1\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\DiskFeature.java"
+++ "b/versions\\1.21.6\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\DiskFeature.java"
@@ -39,15 +39,21 @@ public class DiskFeature extends Feature<DiskFeatureConfig> {
 
 	protected boolean placeBlock(DiskFeatureConfig config, StructureWorldAccess world, Random random, int topY, int bottomY, BlockPos.Mutable pos) {
 		boolean bl = false;
-		BlockState blockState = null;
+		boolean bl2 = false;
 
 		for (int i = topY; i > bottomY; i--) {
 			pos.setY(i);
 			if (config.target().test(world, pos)) {
-				BlockState blockState2 = config.stateProvider().getBlockState(world, random, pos);
-				world.setBlockState(pos, blockState2, Block.NOTIFY_LISTENERS);
-				this.markBlocksAboveForPostProcessing(world, pos);
+				BlockState blockState = config.stateProvider().getBlockState(world, random, pos);
+				world.setBlockState(pos, blockState, Block.NOTIFY_LISTENERS);
+				if (!bl2) {
+					this.markBlocksAboveForPostProcessing(world, pos);
+				}
+
 				bl = true;
+				bl2 = true;
+			} else {
+				bl2 = false;
 			}
 		}
 
