diff --git "a/versions\\1.20.1\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\EndPlacedFeatures.java" "b/versions\\1.21.6\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\EndPlacedFeatures.java"
index 70c38e8..200bb5c 100644
--- "a/versions\\1.20.1\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\EndPlacedFeatures.java"
+++ "b/versions\\1.21.6\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\EndPlacedFeatures.java"
@@ -5,16 +5,19 @@ import net.minecraft.registry.RegistryEntryLookup;
 import net.minecraft.registry.RegistryKey;
 import net.minecraft.registry.RegistryKeys;
 import net.minecraft.registry.entry.RegistryEntry;
+import net.minecraft.server.world.ServerWorld;
 import net.minecraft.util.math.intprovider.UniformIntProvider;
 import net.minecraft.world.gen.YOffset;
 import net.minecraft.world.gen.placementmodifier.BiomePlacementModifier;
 import net.minecraft.world.gen.placementmodifier.CountPlacementModifier;
+import net.minecraft.world.gen.placementmodifier.FixedPlacementModifier;
 import net.minecraft.world.gen.placementmodifier.HeightRangePlacementModifier;
 import net.minecraft.world.gen.placementmodifier.RandomOffsetPlacementModifier;
 import net.minecraft.world.gen.placementmodifier.RarityFilterPlacementModifier;
 import net.minecraft.world.gen.placementmodifier.SquarePlacementModifier;
 
 public class EndPlacedFeatures {
+	public static final RegistryKey<PlacedFeature> END_PLATFORM = PlacedFeatures.of("end_platform");
 	public static final RegistryKey<PlacedFeature> END_SPIKE = PlacedFeatures.of("end_spike");
 	public static final RegistryKey<PlacedFeature> END_GATEWAY_RETURN = PlacedFeatures.of("end_gateway_return");
 	public static final RegistryKey<PlacedFeature> CHORUS_PLANT = PlacedFeatures.of("chorus_plant");
@@ -22,15 +25,19 @@ public class EndPlacedFeatures {
 
 	public static void bootstrap(Registerable<PlacedFeature> featureRegisterable) {
 		RegistryEntryLookup<ConfiguredFeature<?, ?>> registryEntryLookup = featureRegisterable.getRegistryLookup(RegistryKeys.CONFIGURED_FEATURE);
-		RegistryEntry<ConfiguredFeature<?, ?>> registryEntry = registryEntryLookup.getOrThrow(EndConfiguredFeatures.END_SPIKE);
-		RegistryEntry<ConfiguredFeature<?, ?>> registryEntry2 = registryEntryLookup.getOrThrow(EndConfiguredFeatures.END_GATEWAY_RETURN);
-		RegistryEntry<ConfiguredFeature<?, ?>> registryEntry3 = registryEntryLookup.getOrThrow(EndConfiguredFeatures.CHORUS_PLANT);
-		RegistryEntry<ConfiguredFeature<?, ?>> registryEntry4 = registryEntryLookup.getOrThrow(EndConfiguredFeatures.END_ISLAND);
-		PlacedFeatures.register(featureRegisterable, END_SPIKE, registryEntry, BiomePlacementModifier.of());
+		RegistryEntry<ConfiguredFeature<?, ?>> registryEntry = registryEntryLookup.getOrThrow(EndConfiguredFeatures.END_PLATFORM);
+		RegistryEntry<ConfiguredFeature<?, ?>> registryEntry2 = registryEntryLookup.getOrThrow(EndConfiguredFeatures.END_SPIKE);
+		RegistryEntry<ConfiguredFeature<?, ?>> registryEntry3 = registryEntryLookup.getOrThrow(EndConfiguredFeatures.END_GATEWAY_RETURN);
+		RegistryEntry<ConfiguredFeature<?, ?>> registryEntry4 = registryEntryLookup.getOrThrow(EndConfiguredFeatures.CHORUS_PLANT);
+		RegistryEntry<ConfiguredFeature<?, ?>> registryEntry5 = registryEntryLookup.getOrThrow(EndConfiguredFeatures.END_ISLAND);
+		PlacedFeatures.register(
+			featureRegisterable, END_PLATFORM, registryEntry, FixedPlacementModifier.of(ServerWorld.END_SPAWN_POS.down()), BiomePlacementModifier.of()
+		);
+		PlacedFeatures.register(featureRegisterable, END_SPIKE, registryEntry2, BiomePlacementModifier.of());
 		PlacedFeatures.register(
 			featureRegisterable,
 			END_GATEWAY_RETURN,
-			registryEntry2,
+			registryEntry3,
 			RarityFilterPlacementModifier.of(700),
 			SquarePlacementModifier.of(),
 			PlacedFeatures.MOTION_BLOCKING_HEIGHTMAP,
@@ -40,7 +47,7 @@ public class EndPlacedFeatures {
 		PlacedFeatures.register(
 			featureRegisterable,
 			CHORUS_PLANT,
-			registryEntry3,
+			registryEntry4,
 			CountPlacementModifier.of(UniformIntProvider.create(0, 4)),
 			SquarePlacementModifier.of(),
 			PlacedFeatures.MOTION_BLOCKING_HEIGHTMAP,
@@ -49,7 +56,7 @@ public class EndPlacedFeatures {
 		PlacedFeatures.register(
 			featureRegisterable,
 			END_ISLAND_DECORATED,
-			registryEntry4,
+			registryEntry5,
 			RarityFilterPlacementModifier.of(14),
 			PlacedFeatures.createCountExtraModifier(1, 0.25F, 1),
 			SquarePlacementModifier.of(),
