diff --git "a/versions\\1.20.1\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\HugeRedMushroomFeature.java" "b/versions\\1.21.6\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\HugeRedMushroomFeature.java"
index 3efdd58..b066ed5 100644
--- "a/versions\\1.20.1\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\HugeRedMushroomFeature.java"
+++ "b/versions\\1.21.6\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\HugeRedMushroomFeature.java"
@@ -28,22 +28,20 @@ public class HugeRedMushroomFeature extends HugeMushroomFeature {
 					boolean bl6 = bl3 || bl4;
 					if (i >= y || bl5 != bl6) {
 						mutable.set(start, l, i, m);
-						if (!world.getBlockState(mutable).isOpaqueFullCube(world, mutable)) {
-							BlockState blockState = config.capProvider.get(random, start);
-							if (blockState.contains(MushroomBlock.WEST)
-								&& blockState.contains(MushroomBlock.EAST)
-								&& blockState.contains(MushroomBlock.NORTH)
-								&& blockState.contains(MushroomBlock.SOUTH)
-								&& blockState.contains(MushroomBlock.UP)) {
-								blockState = blockState.with(MushroomBlock.UP, i >= y - 1)
-									.with(MushroomBlock.WEST, l < -k)
-									.with(MushroomBlock.EAST, l > k)
-									.with(MushroomBlock.NORTH, m < -k)
-									.with(MushroomBlock.SOUTH, m > k);
-							}
-
-							this.setBlockState(world, mutable, blockState);
+						BlockState blockState = config.capProvider.get(random, start);
+						if (blockState.contains(MushroomBlock.WEST)
+							&& blockState.contains(MushroomBlock.EAST)
+							&& blockState.contains(MushroomBlock.NORTH)
+							&& blockState.contains(MushroomBlock.SOUTH)
+							&& blockState.contains(MushroomBlock.UP)) {
+							blockState = blockState.with(MushroomBlock.UP, i >= y - 1)
+								.with(MushroomBlock.WEST, l < -k)
+								.with(MushroomBlock.EAST, l > k)
+								.with(MushroomBlock.NORTH, m < -k)
+								.with(MushroomBlock.SOUTH, m > k);
 						}
+
+						this.generateStem(world, mutable, blockState);
 					}
 				}
 			}
