diff --git "a/versions\\1.20.1\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\HugeBrownMushroomFeature.java" "b/versions\\1.21.6\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\HugeBrownMushroomFeature.java"
index c076cce..dc77b0b 100644
--- "a/versions\\1.20.1\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\HugeBrownMushroomFeature.java"
+++ "b/versions\\1.21.6\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\HugeBrownMushroomFeature.java"
@@ -26,21 +26,19 @@ public class HugeBrownMushroomFeature extends HugeMushroomFeature {
 				boolean bl6 = bl3 || bl4;
 				if (!bl5 || !bl6) {
 					mutable.set(start, j, y, k);
-					if (!world.getBlockState(mutable).isOpaqueFullCube(world, mutable)) {
-						boolean bl7 = bl || bl6 && j == 1 - i;
-						boolean bl8 = bl2 || bl6 && j == i - 1;
-						boolean bl9 = bl3 || bl5 && k == 1 - i;
-						boolean bl10 = bl4 || bl5 && k == i - 1;
-						BlockState blockState = config.capProvider.get(random, start);
-						if (blockState.contains(MushroomBlock.WEST)
-							&& blockState.contains(MushroomBlock.EAST)
-							&& blockState.contains(MushroomBlock.NORTH)
-							&& blockState.contains(MushroomBlock.SOUTH)) {
-							blockState = blockState.with(MushroomBlock.WEST, bl7).with(MushroomBlock.EAST, bl8).with(MushroomBlock.NORTH, bl9).with(MushroomBlock.SOUTH, bl10);
-						}
-
-						this.setBlockState(world, mutable, blockState);
+					boolean bl7 = bl || bl6 && j == 1 - i;
+					boolean bl8 = bl2 || bl6 && j == i - 1;
+					boolean bl9 = bl3 || bl5 && k == 1 - i;
+					boolean bl10 = bl4 || bl5 && k == i - 1;
+					BlockState blockState = config.capProvider.get(random, start);
+					if (blockState.contains(MushroomBlock.WEST)
+						&& blockState.contains(MushroomBlock.EAST)
+						&& blockState.contains(MushroomBlock.NORTH)
+						&& blockState.contains(MushroomBlock.SOUTH)) {
+						blockState = blockState.with(MushroomBlock.WEST, bl7).with(MushroomBlock.EAST, bl8).with(MushroomBlock.NORTH, bl9).with(MushroomBlock.SOUTH, bl10);
 					}
+
+					this.generateStem(world, mutable, blockState);
 				}
 			}
 		}
