diff --git "a/versions\\1.20.1\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\LargeDripstoneFeature.java" "b/versions\\1.21.6\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\LargeDripstoneFeature.java"
index 8fdd944..7f197fb 100644
--- "a/versions\\1.20.1\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\LargeDripstoneFeature.java"
+++ "b/versions\\1.21.6\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\LargeDripstoneFeature.java"
@@ -35,7 +35,7 @@ public class LargeDripstoneFeature extends Feature<LargeDripstoneFeatureConfig>
 			Optional<CaveSurface> optional = CaveSurface.create(
 				structureWorldAccess, blockPos, largeDripstoneFeatureConfig.floorToCeilingSearchRange, DripstoneHelper::canGenerate, DripstoneHelper::canReplaceOrLava
 			);
-			if (optional.isPresent() && optional.get() instanceof CaveSurface.Bounded) {
+			if (!optional.isEmpty() && optional.get() instanceof CaveSurface.Bounded) {
 				CaveSurface.Bounded bounded = (CaveSurface.Bounded)optional.get();
 				if (bounded.getHeight() < 4) {
 					return false;
