diff --git "a/versions\\1.20.1\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\NetherForestVegetationFeature.java" "b/versions\\1.21.6\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\NetherForestVegetationFeature.java"
index a1a15b0..13ab4a2 100644
--- "a/versions\\1.20.1\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\NetherForestVegetationFeature.java"
+++ "b/versions\\1.21.6\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\NetherForestVegetationFeature.java"
@@ -25,7 +25,7 @@ public class NetherForestVegetationFeature extends Feature<NetherForestVegetatio
 			return false;
 		} else {
 			int i = blockPos.getY();
-			if (i >= structureWorldAccess.getBottomY() + 1 && i + 1 < structureWorldAccess.getTopY()) {
+			if (i >= structureWorldAccess.getBottomY() + 1 && i + 1 <= structureWorldAccess.getTopYInclusive()) {
 				int j = 0;
 
 				for (int k = 0; k < netherForestVegetationFeatureConfig.spreadWidth * netherForestVegetationFeatureConfig.spreadWidth; k++) {
