diff --git "a/versions\\1.20.1\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\TreeFeature.java" "b/versions\\1.21.6\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\TreeFeature.java"
index 99eed6e..265a6f7 100644
--- "a/versions\\1.20.1\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\TreeFeature.java"
+++ "b/versions\\1.21.6\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\TreeFeature.java"
@@ -38,7 +38,7 @@ public class TreeFeature extends Feature<TreeFeatureConfig> {
 		super(codec);
 	}
 
-	private static boolean isVine(TestableWorld world, BlockPos pos) {
+	public static boolean isVine(TestableWorld world, BlockPos pos) {
 		return world.testBlockState(pos, state -> state.isOf(Blocks.VINE));
 	}
 
@@ -70,7 +70,7 @@ public class TreeFeature extends Feature<TreeFeatureConfig> {
 		BlockPos blockPos = (BlockPos)config.rootPlacer.map(rootPlacer -> rootPlacer.trunkOffset(pos, random)).orElse(pos);
 		int m = Math.min(pos.getY(), blockPos.getY());
 		int n = Math.max(pos.getY(), blockPos.getY()) + i + 1;
-		if (m >= world.getBottomY() + 1 && n <= world.getTopY()) {
+		if (m >= world.getBottomY() + 1 && n <= world.getTopYInclusive() + 1) {
 			OptionalInt optionalInt = config.minimumSize.getMinClippedHeight();
 			int o = this.getTopPosition(world, i, blockPos, config);
 			if (o >= i || !optionalInt.isEmpty() && o >= optionalInt.getAsInt()) {
@@ -156,7 +156,7 @@ public class TreeFeature extends Feature<TreeFeatureConfig> {
 
 			return (Boolean)BlockBox.encompassPositions(Iterables.concat(set, set2, set3, set4)).map(box -> {
 				VoxelSet voxelSet = placeLogsAndLeaves(structureWorldAccess, box, set2, set4, set);
-				StructureTemplate.updateCorner(structureWorldAccess, 3, voxelSet, box.getMinX(), box.getMinY(), box.getMinZ());
+				StructureTemplate.updateCorner(structureWorldAccess, Block.NOTIFY_ALL, voxelSet, box.getMinX(), box.getMinY(), box.getMinZ());
 				return true;
 			}).orElse(false);
 		} else {
@@ -227,4 +227,20 @@ public class TreeFeature extends Feature<TreeFeatureConfig> {
 			k++;
 		}
 	}
+
+	public static List<BlockPos> getLeafLitterPositions(TreeDecorator.Generator decoratorGenerator) {
+		List<BlockPos> list = Lists.<BlockPos>newArrayList();
+		List<BlockPos> list2 = decoratorGenerator.getRootPositions();
+		List<BlockPos> list3 = decoratorGenerator.getLogPositions();
+		if (list2.isEmpty()) {
+			list.addAll(list3);
+		} else if (!list3.isEmpty() && ((BlockPos)list2.get(0)).getY() == ((BlockPos)list3.get(0)).getY()) {
+			list.addAll(list3);
+			list.addAll(list2);
+		} else {
+			list.addAll(list2);
+		}
+
+		return list;
+	}
 }
