diff --git "a/versions\\1.20.1\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\TreeConfiguredFeatures.java" "b/versions\\1.21.6\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\TreeConfiguredFeatures.java"
index 0ad4383..02959a6 100644
--- "a/versions\\1.20.1\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\TreeConfiguredFeatures.java"
+++ "b/versions\\1.21.6\\data\\mc_src_extract\\net\\minecraft\\world\\gen\\feature\\TreeConfiguredFeatures.java"
@@ -15,7 +15,7 @@ import net.minecraft.registry.RegistryKey;
 import net.minecraft.registry.RegistryKeys;
 import net.minecraft.registry.entry.RegistryEntryList;
 import net.minecraft.registry.tag.BlockTags;
-import net.minecraft.util.collection.DataPool;
+import net.minecraft.util.collection.Pool;
 import net.minecraft.util.math.Direction;
 import net.minecraft.util.math.intprovider.ConstantIntProvider;
 import net.minecraft.util.math.intprovider.IntProvider;
@@ -43,9 +43,13 @@ import net.minecraft.world.gen.stateprovider.RandomizedIntBlockStateProvider;
 import net.minecraft.world.gen.stateprovider.WeightedBlockStateProvider;
 import net.minecraft.world.gen.treedecorator.AlterGroundTreeDecorator;
 import net.minecraft.world.gen.treedecorator.AttachedToLeavesTreeDecorator;
+import net.minecraft.world.gen.treedecorator.AttachedToLogsTreeDecorator;
 import net.minecraft.world.gen.treedecorator.BeehiveTreeDecorator;
-import net.minecraft.world.gen.treedecorator.CocoaBeansTreeDecorator;
+import net.minecraft.world.gen.treedecorator.CocoaTreeDecorator;
+import net.minecraft.world.gen.treedecorator.CreakingHeartTreeDecorator;
 import net.minecraft.world.gen.treedecorator.LeavesVineTreeDecorator;
+import net.minecraft.world.gen.treedecorator.PaleMossTreeDecorator;
+import net.minecraft.world.gen.treedecorator.PlaceOnGroundTreeDecorator;
 import net.minecraft.world.gen.treedecorator.TrunkVineTreeDecorator;
 import net.minecraft.world.gen.trunk.BendingTrunkPlacer;
 import net.minecraft.world.gen.trunk.CherryTrunkPlacer;
@@ -66,6 +70,9 @@ public class TreeConfiguredFeatures {
 	public static final RegistryKey<ConfiguredFeature<?, ?>> HUGE_RED_MUSHROOM = ConfiguredFeatures.of("huge_red_mushroom");
 	public static final RegistryKey<ConfiguredFeature<?, ?>> OAK = ConfiguredFeatures.of("oak");
 	public static final RegistryKey<ConfiguredFeature<?, ?>> DARK_OAK = ConfiguredFeatures.of("dark_oak");
+	public static final RegistryKey<ConfiguredFeature<?, ?>> PALE_OAK = ConfiguredFeatures.of("pale_oak");
+	public static final RegistryKey<ConfiguredFeature<?, ?>> PALE_OAK_BONEMEAL = ConfiguredFeatures.of("pale_oak_bonemeal");
+	public static final RegistryKey<ConfiguredFeature<?, ?>> PALE_OAK_CREAKING = ConfiguredFeatures.of("pale_oak_creaking");
 	public static final RegistryKey<ConfiguredFeature<?, ?>> BIRCH = ConfiguredFeatures.of("birch");
 	public static final RegistryKey<ConfiguredFeature<?, ?>> ACACIA = ConfiguredFeatures.of("acacia");
 	public static final RegistryKey<ConfiguredFeature<?, ?>> SPRUCE = ConfiguredFeatures.of("spruce");
@@ -84,17 +91,27 @@ public class TreeConfiguredFeatures {
 	public static final RegistryKey<ConfiguredFeature<?, ?>> MANGROVE = ConfiguredFeatures.of("mangrove");
 	public static final RegistryKey<ConfiguredFeature<?, ?>> TALL_MANGROVE = ConfiguredFeatures.of("tall_mangrove");
 	public static final RegistryKey<ConfiguredFeature<?, ?>> CHERRY = ConfiguredFeatures.of("cherry");
-	public static final RegistryKey<ConfiguredFeature<?, ?>> OAK_BEES_0002 = ConfiguredFeatures.of("oak_bees_0002");
+	public static final RegistryKey<ConfiguredFeature<?, ?>> OAK_BEES_0002 = ConfiguredFeatures.of("oak_bees_0002_leaf_litter");
 	public static final RegistryKey<ConfiguredFeature<?, ?>> OAK_BEES_002 = ConfiguredFeatures.of("oak_bees_002");
 	public static final RegistryKey<ConfiguredFeature<?, ?>> OAK_BEES_005 = ConfiguredFeatures.of("oak_bees_005");
 	public static final RegistryKey<ConfiguredFeature<?, ?>> BIRCH_BEES_0002 = ConfiguredFeatures.of("birch_bees_0002");
+	public static final RegistryKey<ConfiguredFeature<?, ?>> BIRCH_BEES_0002_LEAF_LITTER = ConfiguredFeatures.of("birch_bees_0002_leaf_litter");
 	public static final RegistryKey<ConfiguredFeature<?, ?>> BIRCH_BEES_002 = ConfiguredFeatures.of("birch_bees_002");
 	public static final RegistryKey<ConfiguredFeature<?, ?>> BIRCH_BEES_005 = ConfiguredFeatures.of("birch_bees_005");
-	public static final RegistryKey<ConfiguredFeature<?, ?>> FANCY_OAK_BEES_0002 = ConfiguredFeatures.of("fancy_oak_bees_0002");
+	public static final RegistryKey<ConfiguredFeature<?, ?>> FANCY_OAK_BEES_0002 = ConfiguredFeatures.of("fancy_oak_bees_0002_leaf_litter");
 	public static final RegistryKey<ConfiguredFeature<?, ?>> FANCY_OAK_BEES_002 = ConfiguredFeatures.of("fancy_oak_bees_002");
 	public static final RegistryKey<ConfiguredFeature<?, ?>> FANCY_OAK_BEES_005 = ConfiguredFeatures.of("fancy_oak_bees_005");
 	public static final RegistryKey<ConfiguredFeature<?, ?>> FANCY_OAK_BEES = ConfiguredFeatures.of("fancy_oak_bees");
 	public static final RegistryKey<ConfiguredFeature<?, ?>> CHERRY_BEES_005 = ConfiguredFeatures.of("cherry_bees_005");
+	public static final RegistryKey<ConfiguredFeature<?, ?>> OAK_LEAF_LITTER = ConfiguredFeatures.of("oak_leaf_litter");
+	public static final RegistryKey<ConfiguredFeature<?, ?>> DARK_OAK_LEAF_LITTER = ConfiguredFeatures.of("dark_oak_leaf_litter");
+	public static final RegistryKey<ConfiguredFeature<?, ?>> BIRCH_LEAF_LITTER = ConfiguredFeatures.of("birch_leaf_litter");
+	public static final RegistryKey<ConfiguredFeature<?, ?>> FANCY_OAK_LEAF_LITTER = ConfiguredFeatures.of("fancy_oak_leaf_litter");
+	public static final RegistryKey<ConfiguredFeature<?, ?>> FALLEN_OAK_TREE = ConfiguredFeatures.of("fallen_oak_tree");
+	public static final RegistryKey<ConfiguredFeature<?, ?>> FALLEN_JUNGLE_TREE = ConfiguredFeatures.of("fallen_jungle_tree");
+	public static final RegistryKey<ConfiguredFeature<?, ?>> FALLEN_SPRUCE_TREE = ConfiguredFeatures.of("fallen_spruce_tree");
+	public static final RegistryKey<ConfiguredFeature<?, ?>> FALLEN_BIRCH_TREE = ConfiguredFeatures.of("fallen_birch_tree");
+	public static final RegistryKey<ConfiguredFeature<?, ?>> FALLEN_SUPER_BIRCH_TREE = ConfiguredFeatures.of("fallen_super_birch_tree");
 
 	private static TreeFeatureConfig.Builder builder(Block log, Block leaves, int baseHeight, int firstRandomHeight, int secondRandomHeight, int radius) {
 		return new TreeFeatureConfig.Builder(
@@ -110,6 +127,45 @@ public class TreeConfiguredFeatures {
 		return builder(Blocks.OAK_LOG, Blocks.OAK_LEAVES, 4, 2, 0, 2).ignoreVines();
 	}
 
+	private static TreeFeatureConfig.Builder darkOak() {
+		return new TreeFeatureConfig.Builder(
+			BlockStateProvider.of(Blocks.DARK_OAK_LOG),
+			new DarkOakTrunkPlacer(6, 2, 1),
+			BlockStateProvider.of(Blocks.DARK_OAK_LEAVES),
+			new DarkOakFoliagePlacer(ConstantIntProvider.create(0), ConstantIntProvider.create(0)),
+			new ThreeLayersFeatureSize(1, 1, 0, 1, 2, OptionalInt.empty())
+		);
+	}
+
+	private static FallenTreeFeatureConfig.Builder fallenOak() {
+		return fallen(Blocks.OAK_LOG, 4, 7).stumpDecorators(ImmutableList.of(TrunkVineTreeDecorator.INSTANCE));
+	}
+
+	private static FallenTreeFeatureConfig.Builder fallenBirch(int maxLength) {
+		return fallen(Blocks.BIRCH_LOG, 5, maxLength);
+	}
+
+	private static FallenTreeFeatureConfig.Builder fallenJungle() {
+		return fallen(Blocks.JUNGLE_LOG, 4, 11).stumpDecorators(ImmutableList.of(TrunkVineTreeDecorator.INSTANCE));
+	}
+
+	private static FallenTreeFeatureConfig.Builder fallenSpruce() {
+		return fallen(Blocks.SPRUCE_LOG, 6, 10);
+	}
+
+	private static FallenTreeFeatureConfig.Builder fallen(Block log, int minLength, int maxLength) {
+		return new FallenTreeFeatureConfig.Builder(BlockStateProvider.of(log), UniformIntProvider.create(minLength, maxLength))
+			.logDecorators(
+				ImmutableList.of(
+					new AttachedToLogsTreeDecorator(
+						0.1F,
+						new WeightedBlockStateProvider(Pool.<BlockState>builder().add(Blocks.RED_MUSHROOM.getDefaultState(), 2).add(Blocks.BROWN_MUSHROOM.getDefaultState(), 1)),
+						List.of(Direction.UP)
+					)
+				)
+			);
+	}
+
 	private static TreeFeatureConfig.Builder birch() {
 		return builder(Blocks.BIRCH_LOG, Blocks.BIRCH_LEAVES, 5, 2, 0, 2).ignoreVines();
 	}
@@ -141,7 +197,7 @@ public class TreeConfiguredFeatures {
 					1,
 					0,
 					new WeightedListIntProvider(
-						DataPool.<IntProvider>builder().add(ConstantIntProvider.create(1), 1).add(ConstantIntProvider.create(2), 1).add(ConstantIntProvider.create(3), 1).build()
+						Pool.<IntProvider>builder().add(ConstantIntProvider.create(1), 1).add(ConstantIntProvider.create(2), 1).add(ConstantIntProvider.create(3), 1).build()
 					),
 					UniformIntProvider.create(2, 4),
 					UniformIntProvider.create(-4, -3),
@@ -164,6 +220,7 @@ public class TreeConfiguredFeatures {
 			Blocks.ACACIA_SAPLING,
 			Blocks.CHERRY_SAPLING,
 			Blocks.DARK_OAK_SAPLING,
+			Blocks.PALE_OAK_SAPLING,
 			Blocks.MANGROVE_PROPAGULE,
 			Blocks.DANDELION,
 			Blocks.TORCHFLOWER,
@@ -211,6 +268,7 @@ public class TreeConfiguredFeatures {
 			Blocks.FLOWERING_AZALEA,
 			Blocks.MOSS_CARPET,
 			Blocks.PINK_PETALS,
+			Blocks.WILDFLOWERS,
 			Blocks.BIG_DRIPLEAF,
 			Blocks.BIG_DRIPLEAF_STEM,
 			Blocks.SMALL_DRIPLEAF
@@ -292,18 +350,55 @@ public class TreeConfiguredFeatures {
 		BeehiveTreeDecorator beehiveTreeDecorator3 = new BeehiveTreeDecorator(0.02F);
 		BeehiveTreeDecorator beehiveTreeDecorator4 = new BeehiveTreeDecorator(0.05F);
 		BeehiveTreeDecorator beehiveTreeDecorator5 = new BeehiveTreeDecorator(1.0F);
+		PlaceOnGroundTreeDecorator placeOnGroundTreeDecorator = new PlaceOnGroundTreeDecorator(
+			96, 4, 2, new WeightedBlockStateProvider(VegetationConfiguredFeatures.leafLitter(1, 3))
+		);
+		PlaceOnGroundTreeDecorator placeOnGroundTreeDecorator2 = new PlaceOnGroundTreeDecorator(
+			150, 2, 2, new WeightedBlockStateProvider(VegetationConfiguredFeatures.leafLitter(1, 4))
+		);
 		ConfiguredFeatures.register(featureRegisterable, OAK, Feature.TREE, oak().build());
+		ConfiguredFeatures.register(featureRegisterable, DARK_OAK, Feature.TREE, darkOak().ignoreVines().build());
+		ConfiguredFeatures.register(
+			featureRegisterable,
+			PALE_OAK,
+			Feature.TREE,
+			new TreeFeatureConfig.Builder(
+					BlockStateProvider.of(Blocks.PALE_OAK_LOG),
+					new DarkOakTrunkPlacer(6, 2, 1),
+					BlockStateProvider.of(Blocks.PALE_OAK_LEAVES),
+					new DarkOakFoliagePlacer(ConstantIntProvider.create(0), ConstantIntProvider.create(0)),
+					new ThreeLayersFeatureSize(1, 1, 0, 1, 2, OptionalInt.empty())
+				)
+				.decorators(ImmutableList.of(new PaleMossTreeDecorator(0.15F, 0.4F, 0.8F)))
+				.ignoreVines()
+				.build()
+		);
+		ConfiguredFeatures.register(
+			featureRegisterable,
+			PALE_OAK_BONEMEAL,
+			Feature.TREE,
+			new TreeFeatureConfig.Builder(
+					BlockStateProvider.of(Blocks.PALE_OAK_LOG),
+					new DarkOakTrunkPlacer(6, 2, 1),
+					BlockStateProvider.of(Blocks.PALE_OAK_LEAVES),
+					new DarkOakFoliagePlacer(ConstantIntProvider.create(0), ConstantIntProvider.create(0)),
+					new ThreeLayersFeatureSize(1, 1, 0, 1, 2, OptionalInt.empty())
+				)
+				.ignoreVines()
+				.build()
+		);
 		ConfiguredFeatures.register(
 			featureRegisterable,
-			DARK_OAK,
+			PALE_OAK_CREAKING,
 			Feature.TREE,
 			new TreeFeatureConfig.Builder(
-					BlockStateProvider.of(Blocks.DARK_OAK_LOG),
+					BlockStateProvider.of(Blocks.PALE_OAK_LOG),
 					new DarkOakTrunkPlacer(6, 2, 1),
-					BlockStateProvider.of(Blocks.DARK_OAK_LEAVES),
+					BlockStateProvider.of(Blocks.PALE_OAK_LEAVES),
 					new DarkOakFoliagePlacer(ConstantIntProvider.create(0), ConstantIntProvider.create(0)),
 					new ThreeLayersFeatureSize(1, 1, 0, 1, 2, OptionalInt.empty())
 				)
+				.decorators(ImmutableList.of(new PaleMossTreeDecorator(0.15F, 0.4F, 0.8F), new CreakingHeartTreeDecorator(1.0F)))
 				.ignoreVines()
 				.build()
 		);
@@ -357,7 +452,7 @@ public class TreeConfiguredFeatures {
 			JUNGLE_TREE,
 			Feature.TREE,
 			jungle()
-				.decorators(ImmutableList.of(new CocoaBeansTreeDecorator(0.2F), TrunkVineTreeDecorator.INSTANCE, new LeavesVineTreeDecorator(0.25F)))
+				.decorators(ImmutableList.of(new CocoaTreeDecorator(0.2F), TrunkVineTreeDecorator.INSTANCE, new LeavesVineTreeDecorator(0.25F)))
 				.ignoreVines()
 				.build()
 		);
@@ -434,7 +529,7 @@ public class TreeConfiguredFeatures {
 					BlockStateProvider.of(Blocks.OAK_LOG),
 					new BendingTrunkPlacer(4, 2, 0, 3, UniformIntProvider.create(1, 2)),
 					new WeightedBlockStateProvider(
-						DataPool.<BlockState>builder().add(Blocks.AZALEA_LEAVES.getDefaultState(), 3).add(Blocks.FLOWERING_AZALEA_LEAVES.getDefaultState(), 1)
+						Pool.<BlockState>builder().add(Blocks.AZALEA_LEAVES.getDefaultState(), 3).add(Blocks.FLOWERING_AZALEA_LEAVES.getDefaultState(), 1)
 					),
 					new RandomSpreadFoliagePlacer(ConstantIntProvider.create(3), ConstantIntProvider.create(0), ConstantIntProvider.create(2), 50),
 					new TwoLayersFeatureSize(1, 0, 1)
@@ -541,15 +636,51 @@ public class TreeConfiguredFeatures {
 				.ignoreVines()
 				.build()
 		);
-		ConfiguredFeatures.register(featureRegisterable, OAK_BEES_0002, Feature.TREE, oak().decorators(List.of(beehiveTreeDecorator)).build());
+		ConfiguredFeatures.register(
+			featureRegisterable,
+			OAK_BEES_0002,
+			Feature.TREE,
+			oak().decorators(List.of(beehiveTreeDecorator, placeOnGroundTreeDecorator, placeOnGroundTreeDecorator2)).build()
+		);
 		ConfiguredFeatures.register(featureRegisterable, OAK_BEES_002, Feature.TREE, oak().decorators(List.of(beehiveTreeDecorator3)).build());
 		ConfiguredFeatures.register(featureRegisterable, OAK_BEES_005, Feature.TREE, oak().decorators(List.of(beehiveTreeDecorator4)).build());
 		ConfiguredFeatures.register(featureRegisterable, BIRCH_BEES_0002, Feature.TREE, birch().decorators(List.of(beehiveTreeDecorator)).build());
+		ConfiguredFeatures.register(
+			featureRegisterable,
+			BIRCH_BEES_0002_LEAF_LITTER,
+			Feature.TREE,
+			birch().decorators(List.of(beehiveTreeDecorator, placeOnGroundTreeDecorator, placeOnGroundTreeDecorator2)).build()
+		);
 		ConfiguredFeatures.register(featureRegisterable, BIRCH_BEES_002, Feature.TREE, birch().decorators(List.of(beehiveTreeDecorator3)).build());
 		ConfiguredFeatures.register(featureRegisterable, BIRCH_BEES_005, Feature.TREE, birch().decorators(List.of(beehiveTreeDecorator4)).build());
-		ConfiguredFeatures.register(featureRegisterable, FANCY_OAK_BEES_0002, Feature.TREE, fancyOak().decorators(List.of(beehiveTreeDecorator)).build());
+		ConfiguredFeatures.register(
+			featureRegisterable,
+			FANCY_OAK_BEES_0002,
+			Feature.TREE,
+			fancyOak().decorators(List.of(beehiveTreeDecorator, placeOnGroundTreeDecorator, placeOnGroundTreeDecorator2)).build()
+		);
 		ConfiguredFeatures.register(featureRegisterable, FANCY_OAK_BEES_002, Feature.TREE, fancyOak().decorators(List.of(beehiveTreeDecorator3)).build());
 		ConfiguredFeatures.register(featureRegisterable, FANCY_OAK_BEES_005, Feature.TREE, fancyOak().decorators(List.of(beehiveTreeDecorator4)).build());
 		ConfiguredFeatures.register(featureRegisterable, FANCY_OAK_BEES, Feature.TREE, fancyOak().decorators(List.of(beehiveTreeDecorator5)).build());
+		ConfiguredFeatures.register(
+			featureRegisterable, OAK_LEAF_LITTER, Feature.TREE, oak().decorators(ImmutableList.of(placeOnGroundTreeDecorator, placeOnGroundTreeDecorator2)).build()
+		);
+		ConfiguredFeatures.register(
+			featureRegisterable,
+			DARK_OAK_LEAF_LITTER,
+			Feature.TREE,
+			darkOak().ignoreVines().decorators(ImmutableList.of(placeOnGroundTreeDecorator, placeOnGroundTreeDecorator2)).build()
+		);
+		ConfiguredFeatures.register(
+			featureRegisterable, BIRCH_LEAF_LITTER, Feature.TREE, birch().decorators(ImmutableList.of(placeOnGroundTreeDecorator, placeOnGroundTreeDecorator2)).build()
+		);
+		ConfiguredFeatures.register(
+			featureRegisterable, FANCY_OAK_LEAF_LITTER, Feature.TREE, fancyOak().decorators(List.of(placeOnGroundTreeDecorator, placeOnGroundTreeDecorator2)).build()
+		);
+		ConfiguredFeatures.register(featureRegisterable, FALLEN_OAK_TREE, Feature.FALLEN_TREE, fallenOak().build());
+		ConfiguredFeatures.register(featureRegisterable, FALLEN_BIRCH_TREE, Feature.FALLEN_TREE, fallenBirch(8).build());
+		ConfiguredFeatures.register(featureRegisterable, FALLEN_SUPER_BIRCH_TREE, Feature.FALLEN_TREE, fallenBirch(15).build());
+		ConfiguredFeatures.register(featureRegisterable, FALLEN_JUNGLE_TREE, Feature.FALLEN_TREE, fallenJungle().build());
+		ConfiguredFeatures.register(featureRegisterable, FALLEN_SPRUCE_TREE, Feature.FALLEN_TREE, fallenSpruce().build());
 	}
 }
